theano/compile/ops.py

```diff
@@ -488,13 +475,6 @@ def __init__(self, fn, itypes, otypes, infer_shape):
         if self.__infer_shape is not None:
             self.infer_shape = self._infer_shape
 
-    def __eq__(self, other):
-        return (type(self) == type(other) and
-                self.__fn == other.__fn)
-
-    def __hash__(self):
-        return hash(type(self)) ^ hash(self.__fn)
-
```

Comment:
This is a special op. itypes and otypes are only used when building the graph, so they aren't "props" like other props. For now, don't change this op. We are missing what is needed to make it work.